receiver.py

```diff
@@ -180,15 +181,15 @@ def data_lines(self):
                     print("command {}".format(' '.join("0x{:02x}".format(i) for i in cmd)))
                     print("sent a command")
                 self.outbound = []
-            except (OSError, serial.SerialException):
+            except Exception as e:
                 #catch exception if xbee is unplugged, and try to reconnect
-                print("Xbee disconnected!")
+                print("Xbee disconnected! " + str(e))
+                self.xbee = None
                 self.reconnect_xbee()
 
 
     def __exit__(self, type, value, traceback):
         self.xbee = None
-        self.ser.close()
```

Comment:
On resource cleanup, we definitely want to be closing the serial connection. Behaviour of `__exit__` with `with` is similar to how Java's `closeable` works when used in a try-with-resources.